tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php

```diff
@@ -295,4 +298,29 @@ public function testFunctionAnnotation(): void
         $annotation = $reader->getFunctionAnnotation($ref, Fixtures\Annotation\Autoload::class);
         self::assertInstanceOf(Fixtures\Annotation\Autoload::class, $annotation);
     }
+
+    /**
+     * @requires PHP 8.1
+     * @dataProvider provideEnumProperties
+     */
+    public function testAnnotationWithEnum(string $property, Suit $expectedValue): void
+    {
+        $reader = $this->getReader();
+        $ref    = new ReflectionClass(ClassWithEnumAnnotations::class);
+
+        $annotation = $reader->getPropertyAnnotation($ref->getProperty($property), AnnotationWithEnumProperty::class);
+
+        self::assertSame($expectedValue, $annotation->suit);
+    }
+
+    /**
+     * @return list<array{string, Suit}>
+     */
+    public function provideEnumProperties(): array
+    {
+        return [
+            ['annotationWithDefaults', Suit::Hearts],
```

Comment:
Sadly you can't use enums here as PHP will choke on parsing file when running phpunit on older versions

Reply:
I'm surprised the data provider is executed. I'll fix that.

Reply:
PHP breaks on parsing the file, not during execution :)

Reply:
No, the file is parsed just fine as you can tell by the green test suite.

Reply:
Oh, that's actually good to know! I was basing on my last experience with doctrine/common where I hid actual enum usage as much as possible, must have drawn wrong conclusions :)

Reply:
parsing the file containing the enum is what fails. So as long as you don't trigger autoloading for it, you're fine.